core/src/main/scala/chisel3/CompileOptions.scala

```diff
@@ -6,25 +6,37 @@ import scala.language.experimental.macros
 import scala.reflect.macros.blackbox.Context
 
 trait CompileOptions {
-  // Should Record connections require a strict match of fields.
-  // If true and the same fields aren't present in both source and sink, a MissingFieldException,
-  // MissingLeftFieldException, or MissingRightFieldException will be thrown.
+
+  /** Should Record connections require a strict match of fields.
+    *
+    * If true and the same fields aren't present in both source and sink, a MissingFieldException,
+    * MissingLeftFieldException, or MissingRightFieldException will be thrown.
+    */
   val connectFieldsMustMatch: Boolean
-  // When creating an object that takes a type argument, the argument must be unbound (a pure type).
+
+  /** When creating an object that takes a type argument, the argument must be unbound (a pure type). */
   val declaredTypeMustBeUnbound: Boolean
-  // If a connection operator fails, don't try the connection with the operands (source and sink) reversed.
+
+  /** If a connection operator fails, don't try the connection with the operands (source and sink) reversed. */
   val dontTryConnectionsSwapped: Boolean
-  // If connection directionality is not explicit, do not use heuristics to attempt to determine it.
+
+  /** If connection directionality is not explicit, do not use heuristics to attempt to determine it. */
   val dontAssumeDirectionality: Boolean
-  // Check that referenced Data have actually been declared.
+
+  /** Check that referenced Data have actually been declared. */
   val checkSynthesizable: Boolean
-  // Require explicit assignment of DontCare to generate "x is invalid"
+
+  /** Require explicit assignment of DontCare to generate "x is invalid" */
   val explicitInvalidate: Boolean
-  // Should the reset type of Module be a Bool or a Reset
+
+  /** Should the reset type of Module be a Bool or a Reset */
   val inferModuleReset: Boolean
 
   /** If marked true, then any Module which consumes `inferModuleReset=false` must also mix in [[RequireSyncReset]] */
   def migrateInferModuleReset: Boolean = false
+
+  /** Should biconnects use firrtl biconnection semantics (both LHS and RHS must be true to do so) */
+  def firrtlBiConnectSemantics: Boolean = true
```

Comment:
```suggestion
  /** Should connections use firrtl partial connection semantics  */
  def firrtlPartialConnectSemantics: Boolean = true
```
We also issue partial connections for non-Record's too. Also note that for the case of the BiConnect logic below, it's not "both", it's "either". Also there are a couple of other places where we maybe need to change what is being checked, namely https://github.com/chipsalliance/chisel3/blob/231f14e74f112a9f721e774561126b2bd1250039/core/src/main/scala/chisel3/Data.scala#L640 and https://github.com/chipsalliance/chisel3/blob/231f14e74f112a9f721e774561126b2bd1250039/core/src/main/scala/chisel3/Data.scala#L615 which are using `checkSynthesizable` which seems wrong but also perhaps dangerous to fix/change.

Reply:
the `checkSynthesizable` gates there seem legitimate, in that they are gating the `requireIsHardware` checks?

Reply:
oh I see what you mean, *that* usage is legitimate, but then we use that same check for gating a bunch of other stuff unrelated to `checkSynthesizable` flag.